modules/examples/sources/msg-generator/msg-generator-source.c

```diff
@@ -147,6 +147,9 @@ _schedule_wakeup_please(LogSource *s)
   /* not main thread */
   MsgGeneratorSource *self = (MsgGeneratorSource *) s;
 
+  if (!self->suspended)
```

Comment:
I have the same condition inside the event handler `_wakeup`, which is scheduled into the main loop by this function. Looks like some kind of race condition, but it should not be possible.. weird.